tests/test_ptc.py

```diff
@@ -95,50 +99,94 @@ def setUp(self):
+                             len(self.inputNonLinearityResiduals[localDataset.visitMask[ampName]]))
+
+            for i in np.arange(len(localDataset.nonLinearityResiduals[ampName])):
+                self.assertAlmostEqual(localDataset.nonLinearityResiduals[ampName][i],
+                                       self.inputNonLinearityResiduals[i])
+
+        # check entries in returned dataset (should be the same as localDataset after calling the funciton)
         for ampName in self.ampNames:
-            localDataset.rawVars[ampName] = [self.noiseSq + self.c1*mu + self.c2*mu**2 + self.c3*mu**3 for
-                                             mu in localDataset.rawMeans[ampName]]
+            self.assertAlmostEqual('POLYNOMIAL', returnedDataset.ptcFitType[ampName])
+            self.assertAlmostEqual(self.c1, returnedDataset.ptcFitPars[ampName][1])
+            self.assertAlmostEqual(self.gain, returnedDataset.gain[ampName])
+            self.assertAlmostEqual(np.sqrt(self.noiseSq)*self.gain, returnedDataset.noise[ampName])
 
-        config = copy.copy(self.defaultConfig)
-        config.polynomialFitDegree = 3
+            # Nonlinearity fit parameters
+            self.assertAlmostEqual(0.0, localDataset.nonLinearity[ampName][0])
+            self.assertAlmostEqual(self.flux, localDataset.nonLinearity[ampName][1])
+            self.assertAlmostEqual(self.k2NonLinearity, localDataset.nonLinearity[ampName][2])
 
-        task = cpPipe.ptc.MeasurePhotonTransferCurveTask(config=config)
+            # Non-linearity coefficient for linearizer
+            self.assertAlmostEqual(-self.k2NonLinearity/(self.flux**2),
+                                   localDataset.coefficientLinearizeSquared[ampName])
 
-        numberAmps = len(self.ampNames)
-        numberAduValues = config.maxAduForLookupTableLinearizer
-        lookupTableArray = np.zeros((numberAmps, numberAduValues), dtype=np.float32)
+            # Linearity residuals
+            self.assertEqual(len(localDataset.nonLinearityResiduals[ampName]),
+                             len(self.inputNonLinearityResiduals[localDataset.visitMask[ampName]]))
 
-        task.fitPtcAndNonLinearity(localDataset, lookupTableArray, ptcFitType='POLYNOMIAL')
+            for i in np.arange(len(localDataset.nonLinearityResiduals[ampName])):
+                self.assertAlmostEqual(localDataset.nonLinearityResiduals[ampName][i],
+                                       self.inputNonLinearityResiduals[i])
 
-        for ampName in self.ampNames:
-            self.assertAlmostEqual(self.gain, localDataset.gain[ampName])
-            self.assertAlmostEqual(np.sqrt(self.noiseSq)*self.gain, localDataset.noise[ampName])
-            # Linearity residual should be zero
-            self.assertTrue(localDataset.nonLinearityResiduals[ampName].all() == 0)
+    def test_ptcFitQuadAndCubic(self):
+        for order in [2, 3]:
+            self.ptcFitQuadAndCubic(order)
 
     def test_ptcFitAstier(self):
```

Comment:
Almost all of the changes in this function are a duplicate of those in `ptcFitQuadAndCubic`. Could you take this opportunity to combine them into one? It'd be a lot easier to see what was going on, and to make sure they are updated in sync in future!